src/R2.jl

```diff
@@ -136,7 +140,8 @@ function solve!(
     (output.status == :user)
 
   while !done
-    ck .= x .- (∇fk ./ σk)
+    d .= ∇fk .* (T(1) - β) + d .* β  
```

Comment:
```suggestion
    d .= ∇fk .* (T(1) - β) .+ d .* β  
```

Reply:
Maybe you could check if β == 0.

Reply:
If b is zero then d .= ∇fk

Reply:
Yes. It would be cheaper to compute that way.

Reply:
I suggest this then 
    if β == 0
      ck .= x .- (∇fk ./ σk)
    else
      d .= ∇fk .* (T(1) - β) .+ d .* β
      ck .= x .- (d ./ σk)
    end

Reply:
yes, looks good